### PR TITLE
SoundWire: intel: cleanups/fixes before code reorganization

### PR DESCRIPTION
**Pull request by plbossart**
This is a really big cleanup to introduce well defined helpers, prior to a larger code reorganization.

This is relatively invasive to needs to be reviewed carefully, specifically on error flows.

### CODE REVIEW

File: drivers/soundwire/intel.c [hunk 1]
```diff
@@ -1223,6 +1223,174 @@ static int intel_register_dai(struct sdw_intel *sdw)
...
+}
+
+static void intel_check_clock_stop(struct sdw_intel *sdw)
+{
+	struct device *dev = sdw->cdns.dev;
+	bool clock_stop0;
+
+	clock_stop0 = sdw_cdns_is_clock_stop(&sdw->cdns);
+	if (!clock_stop0)
+		dev_err(dev, "%s: invalid configuration, clock was not stopped", __func__);
```
Comment (RanderWang) [inline drivers/soundwire/intel.c]: miss "\n"

File: drivers/soundwire/intel.c [hunk 2]
```diff
@@ -1386,7 +1373,7 @@ int intel_link_startup(struct auxiliary_device *auxdev)
 	}
 
 	/* Initialize shim, controller */
-	ret = intel_init(sdw);
+	ret = intel_link_power_up(sdw);
```
Comment (bardliao) [inline drivers/soundwire/intel.c:1585]: @plbossart We had `intel_link_power_up` and `intel_shim_init` before, but now only `intel_link_power_up` is left.
We call `intel_link_power_down` and `intel_shim_wake` in power_down operations, but only `intel_link_power_up` in power_up operations.
Could you add explanation in the commit message?

File: drivers/soundwire/intel.c [hunk 3]
```diff
@@ -1223,6 +1223,174 @@ static int intel_register_dai(struct sdw_intel *sdw)
...
+
+	if (bus->multi_link) {
+		ret = intel_shim_sync_go(sdw);
+		if (ret < 0) {
+			dev_err(dev, "%s: sync go failed: %d\n", __func__, ret);
+			goto err_interrupt;
+		}
+	}
+	sdw_cdns_check_self_clearing_bits(cdns, __func__,
+					  true, INTEL_MASTER_RESET_ITERATIONS);
```
Comment (bardliao) [inline drivers/soundwire/intel.c:1266]: Do we need to add a `string` that indicates the caller? For example, `static int intel_start_bus(struct sdw_intel *sdw, const char *string)` and call it with `intel_start_bus(sdw, __func__)`

Comment (plbossart) [reply drivers/soundwire/intel.c:1266]: we could do that yes.

File: drivers/soundwire/intel.c (hunk 2 above)
Comment (plbossart) [reply drivers/soundwire/intel.c:1585]: I am not following the comment @bardliao. What is the concern? is this that we have a single operation on power_up and two on power-down?

File: drivers/soundwire/intel.c (hunk 3 above)
Comment (plbossart) [reply drivers/soundwire/intel.c:1266]: Actually we have messages in the caller as well, so it's easy to figure out the sequence. I vote for no change.

File: drivers/soundwire/intel.c (hunk 1 above)
Comment (plbossart) [reply drivers/soundwire/intel.c]: fixed

File: drivers/soundwire/intel.c (hunk 2 above)
Comment (bardliao) [reply drivers/soundwire/intel.c:1585]: Sorry @plbossart please forgive my poor eyes and ignore my comment. For some reason, I thought `intel_shim_init(sdw);` is added in `intel_shim_wake`.

Comment (plbossart) [reply drivers/soundwire/intel.c:1585]: no worries @bardliao

File: drivers/soundwire/intel.c (hunk 3 above)
Comment (plbossart) [reply drivers/soundwire/intel.c:1266]: @bardliao I'll let you deal with this PR, out of time.

**Comment by miRoox** [comment]
SOFCI TEST

**Review by bardliao — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

**Pull request merged**